### ISSUE TITLE
"Failed to parse JSON object as TDLib request: Wrong padding length" When tried to login

### ISSUE THREAD

**Issue by VenkatTeja** [alexander-akhmetov/python-telegram#81]
File "/app/index.py", line 17, in <module>
    tg.login()
  File "/usr/local/lib/python3.8/site-packages/telegram/client.py", line 439, in login
    result.wait(raise_exc=True)
  File "/usr/local/lib/python3.8/site-packages/telegram/utils.py", line 41, in wait
    raise RuntimeError(f'Telegram error: {self.error_info}')
RuntimeError: Telegram error: {'@type': 'error', 'code': 400, 'message': 'Failed to parse JSON object as TDLib request: Wrong padding length', '@extra': {'request_id': 'updateAuthorizationState'}}

Code is running in docker!

Thank you

**Comment by alexander-akhmetov** [comment]
Hi,

On which authorisation state does it fail? It should [print](https://github.com/alexander-akhmetov/python-telegram/blob/0.11.0/telegram/client.py#L435) it to the logs.

**Comment by CatCoderr** [comment]
I have the same problem, here is logs: https://pastebin.com/qAUQr5Ck

**Comment by alexander-akhmetov** [comment]
I see, thanks for the logs. Probably it's related to the encryption key ([parameter](https://github.com/alexander-akhmetov/python-telegram/blob/master/telegram/client.py#L30) `database_encryption_key`). 

While it's not fixed in `python-telegram`, can you try to make the database encryption key string longer? Probably you'll need to remove existing `tdlib` database. By default `python-telegram` sets it to `/tmp/.tdlib_files/`

**Comment by steven-chau** [comment]
I have the exact same problem! Same error. Same stacktrace.

I have tried a long `database_encryption_key` and a short one. No luck :(

Perhaps the protocol has changed? The underlying `tdLib` library needs to be upgraded?

**Comment by jonasborges** [comment]
> I see, thanks for the logs. Probably it's related to the encryption key ([parameter](https://github.com/alexander-akhmetov/python-telegram/blob/master/telegram/client.py#L30) `database_encryption_key`).
> 
> While it's not fixed in `python-telegram`, can you try to make the database encryption key string longer? Probably you'll need to remove existing `tdlib` database. By default `python-telegram` keeps it in `/tmp/.tdlib_files/`
> 
> ```python
> tg = Telegram(
>     database_encryption_key='...',
> )
> ```

It seems to be the parameter name being sent.

The example provided by [tdlib](https://github.com/tdlib/td/blob/v1.6.0/example/python/tdjson_example.py#L115) uses **key** as parameter for `checkDatabaseEncryptionKey` messages. 

```python
{'@type': 'checkDatabaseEncryptionKey', 'key': '...'}
```

While python-telegram is using **[encryption_key](https://github.com/alexander-akhmetov/python-telegram/blob/0.11.0/telegram/client.py#L478)**.
The [documentation](https://core.telegram.org/tdlib/docs/classtd_1_1td__api_1_1check_database_encryption_key.html#a8d01721f2c7c5f0f2eeac997db78b0ee) also mention **encryption_key**.

So, I made it work by changing the key name from `encryption_key` to `key`. I'm not sure if this is the proper solution but works for now.

**Comment by alexander-akhmetov** [comment]
> The example provided by tdlib uses key as parameter for checkDatabaseEncryptionKey messages.

Hm, interesting! I'm not sure if it's a mistake or they are interchangeable. I will try to look into that.

I also found that in [td_api.tl](https://github.com/tdlib/td/blob/v1.6.0/td/generate/scheme/td_api.tl#L3032) `encryption_key` has type `bytes` and in the [documentation](https://core.telegram.org/tdlib/docs/classtd_1_1td__api_1_1check_database_encryption_key.html#ac482ecf8b42398fca13bc54a545c8e2e) it is `string`. They mentioned in [one of the issues](https://github.com/tdlib/td/issues/118#issuecomment-370159221) that bytes should be sent as base64 encoded string. The error original error message (`Wrong padding length`)  is also related to [base64](https://github.com/tdlib/td/blob/92c9149e30318ac83baeb194ba9fcacf2bea3588/tdutils/td/utils/base64.cpp#L88). 

With base64 encoded string it seems to be working, but I don't know yet how to check if tdlib is using the correct and decoded from base64 key, or just the passed string:
```python
    def _send_encryption_key(self) -> AsyncResult:
        logger.info('Sending encryption key')

        key = self._database_encryption_key
        if isinstance(key, str):
            key = key.encode()

        data = {
            '@type': 'checkDatabaseEncryptionKey',
            'encryption_key': base64.b64encode(key).decode(),
        }

        return self._send_data(data, result_id='updateAuthorizationState')
```

Can somebody try to use the default key from the python-telegram examples (`database_encryption_key='changeme1234'`) and an empty key (`database_encryption_key=''`), please?

**Comment by alexander-akhmetov** [comment]
> I asked a question in tdlib repository about base64 and difference between key and encryption_key: tdlib/td#970

Yes, it has bytes type and should be encoded using base64

**Comment by alexander-akhmetov** [comment]
Released `0.12.0`, if you still see the problem, please re-open the issue
